daft/analytics.py

```diff
@@ -62,21 +64,38 @@ def _build_segment_batch_payload(
     }
 
 
+def _enable_analytics() -> None:
+    global _ANALYTICS_ACTIVE
+    _ANALYTICS_ACTIVE = True
+
+
 def _post_segment_track_endpoint(payload: dict[str, Any]) -> None:
     """Posts a batch of JSON data to Segment"""
-    req = urllib.request.Request(
-        _SEGMENT_BATCH_ENDPOINT,
-        method="POST",
-        headers={
-            "Content-Type": "application/json",
-            "User-Agent": "daft-analytics",
-            "Authorization": f"Basic {base64.b64encode(f'{_WRITE_KEY}:'.encode()).decode('utf-8')}",
-        },
-        data=json.dumps(payload).encode("utf-8"),
-    )
-    resp = urllib.request.urlopen(req)
-    if resp.status != 200:
-        raise RuntimeError(f"HTTP request to segment returned status code: {resp.status}")
+    global _ANALYTICS_ACTIVE
+    if _ANALYTICS_ACTIVE:
```

Comment:
This takes care of publishing, but we should also probably stop adding to the buffer? Otherwise the buffer is going to keep growing even though we aren't going to make any calls to the analytics endpoint

Reply:
Updated to stop appending to buffer is `self._is_active` is `False`.